flowclient/flowclient/client.py

```diff
@@ -1099,3 +1103,36 @@ def joined_spatial_aggregate(
         "locations": locations,
         "metric": metric,
     }
+
+
+def aggregate_network_objects(
+    *, total_network_objects: dict, statistic: str, period: str
+) -> dict:
+    """
+    Return query spec for aggregate network objects
+
+    Parameters
+    ----------
+    total_network_objects : dict
+        TotalNetworkObjects query result
+    statistic : str
+        Statistic type
+    period : str
+        Period type
+
+    Returns
+    -------
+    dict
+        Query specification for an aggregated network objects query
+    """
+    total_network_objs = total_network_objects
+
+    if "period" not in total_network_objs:
+        total_network_objs["period"] = period
```

Comment:
Why is this here?

Reply:
Because without it the tests error as follows:

> E           flowclient.client.FlowclientConnectionError: Something went wrong. API returned with status code 400.
> Error message: 'Internal flowmachine server error: could not create query object using query schema.
> The original error was: '__init__() missing 1 required keyword-only argument: 'period'''.
> Payload: {'orig_error_msg': "__init__() missing 1 required keyword-only argument: 'period'",
> 'params': {'period': 'day', 'query_kind': 'aggregate_network_objects', 'statistic': 'median', 
> 'total_network_objects': {'aggregation_unit': 'admin3', 'end_date': '2016-01-02', 'query_kind': 'total_network_objects', 'start_date': '2016-01-01'}}}

Reply:
Solution is to pass in valid json for the nested query, rather than silently patch it up here I think.. if you look at the total_network_objects function (which is what would be supplying the value for the total_network_objs argument), it has a default value of ‘day’ for the period.